UPGRADE-4.0.md

```diff
@@ -18,6 +18,16 @@ DependencyInjection
 
  * Using unsupported options to configure service aliases raises an exception.
 
+ * Resetting a private shared service from outside of the container with the
+   `set()` method of the `Container` class is no longer supported and will raise
+   an exception.
```

Comment:
I think we need to be more precise here. It won't raise any new exception, it will just tell the dev that the service does not exist.